### PR TITLE
add kubernetes module, example and CI

### PR DESCRIPTION
**Pull request by penpyt**
# Goal

Add AKS Kubernetes to `scalar-terraform`

## Tasks

- [x] add terraform Kubernetes module
- [x] add example Kubernetes
- [x] add CI
- [x] Tested With Kelpie

## Reference

https://scalar-labs.atlassian.net/browse/DLT-6948

### CODE REVIEW

**Comment by ymorimo** [comment]
@penpyt A Kubernetes service was successfully deployed just like `scakar-k8s`. But I'm not sure how it it supposed to be used. The `operation` directory in scalar-k8s should be used?

**Comment by feeblefakie** [comment]
@penpyt Is it just a copy from scalar-k8s repo ? or have you updated something for this repo ?

**Comment by penpyt** [comment]
> @penpyt Is it just a copy from scalar-k8s repo ? or have you updated something for this repo ?

yes only copy from `scalar-k8s`

**Comment by penpyt** [comment]
> @penpyt A Kubernetes service was successfully deployed just like `scakar-k8s`. But I'm not sure how it it supposed to be used. The `operation` directory in scalar-k8s should be used?

yes you need to use the `scalar-k8s` repo to deploy the application on top of kubernetes, please see https://github.com/scalar-labs/scalar-k8s/pull/63 where I did adjust the documents

File: examples/azure/kubernetes/main.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module "kubernetes" {
```
Comment (tei-k) [inline examples/azure/kubernetes/main.tf:1]: As with the other modules, should add the following.
```
# source = "git@github.com:scalar-labs/scalar-terraform.git//modules/azure/kubernetes?ref=v1.0.0"
```

Comment (penpyt) [reply examples/azure/kubernetes/main.tf:1]: added in https://github.com/scalar-labs/scalar-terraform/pull/182/commits/214953f707598fa6fffbd39cb3dcb4dbe7f08f63

File: modules/azure/kubernetes/azure.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+terraform {
+  required_version = ">= 0.12.20"
+}
+
```
Comment (tei-k) [inline modules/azure/kubernetes/azure.tf:4]: Separate the terraform version part into versions.tf file.

Comment (penpyt) [reply modules/azure/kubernetes/azure.tf]: separated in https://github.com/scalar-labs/scalar-terraform/pull/182/commits/9809c62a0a7e0188f36c6d7a6c7d617fe7804996

File: test/modules/test_key.pub [hunk 3]
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCyvESRzWdz4/Q/4CHO9pJfH8SKOBnz9o99bnc8vpLnDqTFQN1MozkXM2iZcI6PlLIrrUMVRNjby5KF0R7imexitzchnTfqWe5H1/sqjgJlKrL6/Ack2OMeOetrsQzcQ84j1Je2esAZlH5n1RmRdXEWzt14M7CD9gDvX8j5NlMeplN7uHbY4e/SbTnWmtvug65RZf71lr31DBuJjWUf70kxyYoUO5J9Dcq/OFpjETlHJE54MpWFbSOYkJB1HWcBeuFT7N12VEV2J+EFECJ+zItG5yO9/wl1txg6zY9qvyw7qA7DVH4jZHfGBLs+DsBVAz7RDDCvXWt8NTkTPQYKaG4b test@test.local
```
Comment (tei-k) [inline test/modules/test_key.pub:1]: Do we need the contents of the file?

Comment (penpyt) [reply test/modules/test_key.pub:1]: yes otherwise the CI failed in https://github.com/scalar-labs/scalar-terraform/runs/978698972

Comment (tei-k) [reply test/modules/test_key.pub:1]: Thanks, Got it. 👍

File: test/modules/azuredeploy/kubernetes/terraform.tfvars [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Required
+kubernetes_cluster_availability_zones = ["1", "2", "3"]
+
+# Optional
```
Comment (tei-k) [inline test/modules/azuredeploy/kubernetes/terraform.tfvars:4]: If we want to test with the default value, I think don't need to specify value in optinal.

Comment (penpyt) [reply test/modules/azuredeploy/kubernetes/terraform.tfvars:1]: removed in https://github.com/scalar-labs/scalar-terraform/pull/182/commits/de42d5bb899eaaa20581ad042d8bd92c7c44dab7

Comment (tei-k) [reply test/modules/azuredeploy/kubernetes/terraform.tfvars:1]: Oh! sorry!
I mean optional variables `kubernetes_cluster` , `kubernetes_default_node_pool` , `kubernetes_scalar_apps_pool` .
`kubernetes_cluster_availability_zones` seems to be required.
e.g.
https://github.com/scalar-labs/scalar-terraform/blob/master/test/modules/azuredeploy/scalardl/terraform.tfvars

Comment (penpyt) [reply test/modules/azuredeploy/kubernetes/terraform.tfvars:1]: @tei-k one default var wasn't set for kubernetes_cluster_availability_zones, all of them are Optional

Comment (tei-k) [reply test/modules/azuredeploy/kubernetes/terraform.tfvars:1]: OK! I confirmed that default value added in `vars.tf`.

The optional variables should use the default even if don't specify them, so I just thought it was unnecessary to specify them here.

**Pull request merged**